### PR TITLE
Normalize the usage of stats collectors

### PR DESCRIPTION
**Pull request by lavoiesl**
- Remove Metric, superseeded by the various collectors
- Make collector type private
   Specific collectors and the properties are still public, so the usage doesn't change.
- Inline standalone functions inside the relevant metrics
   Removes duplication and normalizes the usage

### CODE REVIEW

File: statsd/statsd.go [hunk 1]
```diff
@@ -64,36 +64,6 @@ func NewBackend(impl, addr, prefix string, tags ...string) (Backend, error) {
 	return b, errors.WithStack(err)
 }
 
-// Gauge measures the value of a metric at a particular time.
```
Comment (inf-rno) [inline statsd/statsd.go:67]: Getting rid of these package level functions which expose the singleton underneath is a breaking change. I know of  a few different projects that use this API. Why was this removed?

**Review by inf-rno — COMMENTED** [review]
This PR breaks usage of existing API in prod for multiple repos. It was a while back, but when I wrote this, the idea of the Singleton implementation instead of passing around the statsd object is so that you can just `statsd.Count()` and just have it work.

File: statsd/statsd.go (hunk 1 above)
Comment (lavoiesl) [reply statsd/statsd.go:67]: The `currentBackend` singleton is still available, but the emitting of ad-hoc events was intentionally removed, to promote the use of defined metrics.

It simplifies the goose code base and helps avoid mistakes in magic strings in the clients.

e.g. Instead of this:
```go
statsd.Gauge(ctx, "event.concurrency", 10, nil, 0)
```

Use this:
```go
var (
  EventConcurrency = &statsd.Gaugor{Name: "event.concurrency"}
)

# ...

EventConcurrency.Gauge(ctx, 10)
```

---

Is it indeed a breaking change and Goose would benefit from proper semver, but I recommend updating the client's code to the recommended API.

Comment (inf-rno) [reply statsd/statsd.go:67]: I understand what you're trying to do here. But I'm not sure I agree that removing these lines here is making the code base much simpler. Seems like the main thing the collector struct is abstracting is the metric name to prevent magic string typos etc. If these metrics (and most metrics are like this) are only emitted in a single spot in the code base then it doesn't buy us much either. 

Do we know how many clients use this library and was there some sort of consensus towards this direction? Otherwise, a silent breaking change is a good way for a library to get pinned to an old version or forked which seems counter productive to what you're after. I recommend leaving these in for codebase that are already in production and for use cases that gain no benefit from the collector abstraction.
